TypeError invoking fixture when redis is not installed

### What action do you want to perform

Simply use the plugin.

### What are the results

```
$ cat > test_pytest_redis.py
__requires__ = ['pytest', 'pytest-redis==1.2.0']

def test_fixture(redisdb):
    pass
$ rwt -- -m pytest test_pytest_redis.py
Collecting pytest-redis==1.2.0
  Using cached pytest_redis-1.2.0-py2.py3-none-any.whl
Collecting redis (from pytest-redis==1.2.0)
  Using cached redis-2.10.5-py2.py3-none-any.whl
Collecting pytest>=3.0.0 (from pytest-redis==1.2.0)
  Using cached pytest-3.0.7-py2.py3-none-any.whl
Collecting mirakuru>=0.2 (from pytest-redis==1.2.0)
  Using cached mirakuru-0.8.2-py2.py3-none-any.whl
Collecting port-for>=0.3.1 (from pytest-redis==1.2.0)
Collecting py>=1.4.29 (from pytest>=3.0.0->pytest-redis==1.2.0)
  Using cached py-1.4.33-py2.py3-none-any.whl
Collecting setuptools (from pytest>=3.0.0->pytest-redis==1.2.0)
  Using cached setuptools-34.3.3-py2.py3-none-any.whl
Collecting psutil>=4.0.0 (from mirakuru>=0.2->pytest-redis==1.2.0)
Collecting packaging>=16.8 (from setuptools->pytest>=3.0.0->pytest-redis==1.2.0)
  Using cached packaging-16.8-py2.py3-none-any.whl
Collecting appdirs>=1.4.0 (from setuptools->pytest>=3.0.0->pytest-redis==1.2.0)
  Using cached appdirs-1.4.3-py2.py3-none-any.whl
Collecting six>=1.6.0 (from setuptools->pytest>=3.0.0->pytest-redis==1.2.0)
  Using cached six-1.10.0-py2.py3-none-any.whl
Collecting pyparsing (from packaging>=16.8->setuptools->pytest>=3.0.0->pytest-redis==1.2.0)
  Using cached pyparsing-2.2.0-py2.py3-none-any.whl
Installing collected packages: redis, py, six, pyparsing, packaging, appdirs, setuptools, pytest, psutil, mirakuru, port-for, pytest-redis
Successfully installed appdirs-1.4.3 mirakuru-0.8.2 packaging-16.8 port-for-0.3.1 psutil-5.2.1 py-1.4.33 pyparsing-2.2.0 pytest-3.0.7 pytest-redis-1.2.0 redis-2.10.5 setuptools-34.3.3 six-1.10.0
====================================== test session starts =======================================
platform darwin -- Python 3.6.1, pytest-3.0.7, py-1.4.33, pluggy-0.4.0
rootdir: /Users/jaraco, inifile:
plugins: redis-1.2.0
collected 1 items 

test_pytest_redis.py E

============================================= ERRORS =============================================
_________________________________ ERROR at setup of test_fixture _________________________________

request = <SubRequest 'redisdb' for <Function 'test_fixture'>>

    @pytest.fixture
    def redisdb_factory(request):
        """
            Connection fixture for pytest-redis.
    
            #. Load required process fixture.
            #. Get redis module and config.
            #. Connect to redis.
            #. Flush database after tests.
    
            :param FixtureRequest request: fixture request object
            :rtype: redis.client.Redis
            :returns: Redis client
            """
>       proc_fixture = request.getfixturevalue(process_fixture_name)

/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/rwt-o41ay1dl/pytest_redis/factories.py:195: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/rwt-o41ay1dl/pytest_redis/factories.py:156: in redis_proc_fixture
    cv_result = compare_version(redis_version, REQUIRED_VERSION)
/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/rwt-o41ay1dl/pytest_redis/factories.py:71: in compare_version
    return cmp_v(normalize(version1), normalize(version2))
/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/rwt-o41ay1dl/pytest_redis/factories.py:67: in normalize
    return [int(x) for x in re.sub(r'(\.0+)*$', '', v).split(".")]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

pattern = '(\\.0+)*$', repl = '', string = None, count = 0, flags = 0

    def sub(pattern, repl, string, count=0, flags=0):
        """Return the string obtained by replacing the leftmost
        non-overlapping occurrences of the pattern in string by the
        replacement repl.  repl can be either a string or a callable;
        if a string, backslash escapes in it are processed.  If it is
        a callable, it's passed the match object and must return
        a replacement string to be used."""
>       return _compile(pattern, flags).sub(repl, string, count)
E       TypeError: expected string or bytes-like object

/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/re.py:191: TypeError
------------------------------------- Captured stderr setup --------------------------------------
/bin/sh: /usr/bin/redis-server: No such file or directory
==================================== 1 error in 0.10 seconds =====================================
```

### What are the expected results

1 test should pass or be skipped if redis isn't installed.

--- a/src/pytest_redis/factories.py
+++ b/src/pytest_redis/factories.py
@@ -52,6 +52,12 @@ class RedisUnsupported(Exception):
     pass
 
 
+class RedisMisconfigured(Exception):
+    """Exception raised when the redis_exec points to non existing file."""
+
+    pass
+
+
 def compare_version(version1, version2):
     """
     Compare two version numbers.
@@ -135,6 +141,24 @@ def redis_proc(
             if compression is None else compression
         rdbchecksum = config['rdbchecksum'] if checksum is None else checksum
 
+        version_string = os.popen('{0} --version'.format(redis_exec)).read()
+        if not version_string:
+            raise RedisMisconfigured(
+                'Bad path to redis_exec is given:'
+                ' {0} not exists or wrong program'.format(
+                    redis_exec
+                )
+            )
+
+        redis_version = extract_version(version_string)
+        cv_result = compare_version(redis_version, REQUIRED_VERSION)
+        if redis_version and cv_result < 0:
+            raise RedisUnsupported(
+                'Your version of Redis is not supported. '
+                'Consider updating to Redis {0} at least. '
+                'The currently installed version of Redis: {1}.'
+                .format(REQUIRED_VERSION, redis_version))
+
         redis_executor = RedisExecutor(
             executable=redis_exec,
             databases=db_count or config['db_count'],
@@ -150,16 +174,6 @@ def redis_proc(
             port=get_port(port) or get_port(config['port']),
             timeout=60,
         )
-        redis_version = extract_version(
-            os.popen('{0} --version'.format(redis_exec)).read()
-        )
-        cv_result = compare_version(redis_version, REQUIRED_VERSION)
-        if redis_version and cv_result < 0:
-            raise RedisUnsupported(
-                'Your version of Redis is not supported. '
-                'Consider updating to Redis {0} at least. '
-                'The currently installed version of Redis: {1}.'
-                .format(REQUIRED_VERSION, redis_version))
         redis_executor.start()
         request.addfinalizer(redis_executor.stop)
 

--- a/tests/test_redis.py
+++ b/tests/test_redis.py
@@ -7,7 +7,7 @@ import mock
 from pytest_redis import factories
 from pytest_redis.factories import (
     RedisUnsupported, extract_version, compare_version,
-)
+    RedisMisconfigured)
 
 
 def test_redis(redisdb):
@@ -87,3 +87,15 @@ def test_compare_version(versions, result):
 def test_extract_version(text, result):
     """Check if the version extracction works correctly."""
     assert extract_version(text) == result
+
+
+redis_not_existing = factories.redis_proc(
+    executable='/not/redis/here/redis-server',
+    port=None
+)
+
+
+def test_not_existing_redis(request):
+    """Check handling of misconfigured redis executable path."""
+    with pytest.raises(RedisMisconfigured):
+        request.getfixturevalue('redis_not_existing')

